tests/Aspire.Hosting.Tests/ProjectResourceTests.cs

```diff
@@ -14,6 +14,47 @@ namespace Aspire.Hosting.Tests;
 
 public class ProjectResourceTests
 {
+    [Fact]
+    public async Task AddProjectWithInvalidLaunchSettingsShouldThrowSpecificError()
+    {
+        var projectDetails = await PrepareProjectWithMalformedLaunchSettingsAsync();
+
+        var ex = Assert.Throws<DistributedApplicationException>(() =>
+        {
+            var appBuilder = CreateBuilder();
+            appBuilder.AddProject("project", projectDetails.ProjectFilePath);
+        });
+
+        var expectedMessage = $"Failed to get effective launch profile for project resource 'project'. There is malformed JSON in the project's launch settings file at '{projectDetails.LaunchSettingsFilePath}'.";
+        Assert.Equal(expectedMessage, ex.Message);
+
+        async static Task<(string ProjectFilePath, string LaunchSettingsFilePath)> PrepareProjectWithMalformedLaunchSettingsAsync()
+        {
+            var csProjContent = """
+                                <Project Sdk="Microsoft.NET.Sdk.Web">
+                                <!-- Not a real project, just a stub for testing -->
+                                </Project>
+                                """;
+
+            var launchSettingsContent = """
+                                        this { is } { mal formed! >
+                                        """;
+
+            var projectDirectoryPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            var projectFilePath = Path.Combine(projectDirectoryPath, "Project.csproj");
+            var propertiesDirectoryPath = Path.Combine(projectDirectoryPath, "Properties");
+            var launchSettingsFilePath = Path.Combine(propertiesDirectoryPath, "launchSettings.json");
+
+            Directory.CreateDirectory(projectDirectoryPath);
```

Comment:
```suggestion
            var projectDirectoryPath = Directory.CreateTempSubdirectory().FullName;
            var projectFilePath = Path.Combine(projectDirectoryPath, "Project.csproj");
            var propertiesDirectoryPath = Path.Combine(projectDirectoryPath, "Properties");
            var launchSettingsFilePath = Path.Combine(propertiesDirectoryPath, "launchSettings.json");

```

Using CreateTempSubdirectory is easier.